### ISSUE TITLE
Don't match URL that ends punctuation marks

### ISSUE THREAD

**Issue by Reon90** [kevva/url-regex#23]
Ex: http://facebook.com/text,

**Comment by kevva** [comment]
Why? That isn't a valid URL.

**Comment by Reon90** [comment]
Sorry, i corrected the title.
Your regexp thinks that this 'http://facebook.com/text,' is the correct URL

**Comment by kevva** [comment]
Are you using the `exact` option?

**Comment by Reon90** [comment]
No, I think if URL ends with punctuation marks, it shouldn't get into URL.
For ex, i use it to get url from message 'It's cool video https://www.youtube.com/watch?v=25OUFtdno8U, Did you see it?'
In this case i got 'https://www.youtube.com/watch?v=25OUFtdno8U,'

**Comment by radiovisual** [comment]
@Reon90 , you are referring to this behavior?:

```js
var urlRegExp = require("url-regex");

var string = 'It\'s cool video https://www.youtube.com/watch?v=25OUFtdno8U, Did you see it?';
string.match(urlRegExp());
//=> ['https://www.youtube.com/watch?v=25OUFtdno8U,']
```

Where the returned array in `string.match(urlRegExp())` is returning a url with the comma at the end?

**Comment by radiovisual** [comment]
I am not an expert on url patterns, but I know that commas, and other punctuation are sometimes allowed in URLs (or URIs):

*via http://stackoverflow.com/a/1547940/3960969:*
> In general URIs as defined by RFC 3986 (see Section 2: Characters) may contain any of the following characters: ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~:/?#[]@!$&'()*+,;=

So it could be that your url with the trailing comma is actually considered a valid url, and url-regex has no way of knowing where the url ends and when the punctuation begins.

**Comment by kevva** [comment]
Thanks for clarifying @radiovisual.

**Comment by gajus** [comment]
Would be at least nice to have this as an option.

I have not ever seen a single URL in wild that would end with a comma or dot. Meanwhile, there are a lot of occurrences in text where comma, dot will appear at the end of a URL.

**Comment by niftylettuce** [comment]
This is fixed in `url-regex-safe` v1.0.0 at https://github.com/niftylettuce/url-regex-safe.

Feel free to use this as a replacement!
